chia/plotting/util.py

```diff
@@ -119,13 +121,20 @@ def get_filenames(directory: Path) -> List[Path]:
         return []
     all_files: List[Path] = []
     try:
-        for child in directory.iterdir():
-            if not child.is_dir():
-                # If it is a file ending in .plot, add it - work around MacOS ._ files
-                if child.suffix == ".plot" and not child.name.startswith("._"):
-                    all_files.append(child)
-            else:
-                log.debug(f"Not checking subdirectory {child}, subdirectories not added by default")
+        if recursive:
+            all_files = [
+                child
+                for child in directory.rglob("*.plot")
+                if child.is_file() and child.suffix == ".plot" and not child.name.startswith("._")
+            ]
+        else:
+            for child in directory.iterdir():
+                if not child.is_dir():
+                    # If it is a file ending in .plot, add it - work around MacOS ._ files
+                    if child.suffix == ".plot" and not child.name.startswith("._"):
+                        all_files.append(child)
+                else:
+                    log.debug(f"Not checking subdirectory {child}, subdirectories not added by default")
```

Comment:
I guess this logging is the only reason we need two separate mechanisms here, otherwise it is just `.rglob()` vs. `.glob()`.  `:|`  Though this message should probably reference the new feature now.  And maybe it should log that it is recursing when doing so.  It is a debug message after all.

```suggestion
                    log.debug(f"Subdirectory not searched for plots, see setting harvester / recursive_plot_scan: {child}")
```

Note that bringing these two paths back to one might be done via https://docs.python.org/3.10/library/os.html#os.walk.  It has a really weird but useful interface.  As you iterate it you can modify the results such as removing directories from its list to tell it to not traverse them.

Or, just use this function recursively on recursive subdirectory configs.

Reply:
Makes sense, see bf0cd5b5cebb3874eac2689cb913d0d30e045e29. I merged the lookups and changed the log a bit. I guess since its `DEBUG` anyway its not super important to keep this "warning".